xapian-applications/omega/index_file.cc

```diff
@@ -551,44 +552,45 @@ bool index_update_entry(const string & urlterm,DirectoryIterator &d){
 
 void
 index_mimetype(const string & file, const string & urlterm, const string & url,
-	       const string & ext, string &mimetype, DirectoryIterator &d,
-	       Xapian::Document & newdocument,
-	       string record)
+	       const string & ext, string &mimetype, string path_term,
+               DirectoryIterator &d, string record)
 {
     string context(file, root.size(), string::npos);
 
     time_t last_altered = use_ctime ? d.get_ctime() : d.get_mtime();
 
     Xapian::docid did = 0;
     if (index_check_existing(urlterm, last_altered, did))
-	return;
+	   return;
 
     if (!retry_failed) {
-	// We only store and check the mtime (last modified) - a change to the
-	// metadata won't generally cause a previous failure to now work
-	// (FIXME: except permissions).
-	time_t failed_last_mod;
-	off_t failed_size;
-	if (failed.contains(urlterm, failed_last_mod, failed_size)) {
-	    if (d.get_mtime() <= failed_last_mod &&
-		d.get_size() == failed_size) {
-		if (verbose)
-		    cout << "failed to extract text on earlier run" << endl;
-		return;
-	    }
-	    // The file has changed, so remove the entry for it.  If it fails
-	    // again on this attempt, we'll add a new one.
-	    failed.del(urlterm);
-	}
+    	// We only store and check the mtime (last modified) - a change to the
+    	// metadata won't generally cause a previous failure to now work
+    	// (FIXME: except permissions).
+    	time_t failed_last_mod;
+    	off_t  failed_size;
+    	if (failed.contains(urlterm, failed_last_mod, failed_size)) {
+    	    if (d.get_mtime() <= failed_last_mod &&
+    		d.get_size() == failed_size) {
+    		if (verbose)
+    		    cout << "failed to extract text on earlier run" << endl;
+    		return;
+    	    }
+    	    // The file has changed, so remove the entry for it.  If it fails
+    	    // again on this attempt, we'll add a new one.
+    	    failed.del(urlterm);
+    	}
     }
 
+    // if the ctime has changed but the mtime is unchanged, we can just
+    // update the existing Document and avoid having to re-extract text, etc.
     if(use_ctime && d.get_ctime()!=d.get_mtime()){
-        // if the ctime has changed but the mtime is unchanged, we can just
-        // update the existing Document and avoid having to re-extract text, etc.
-        // if(index_update_entry(urlterm,d))
-            // return;
+        if(index_update_entry(urlterm,d))
```

Comment:
Again, this doesn't belong in this PR so hopefully you can pull it out.

Reply:
You are right, I'm going to delete this modification and see if I can move it to another pull request related to ticket 632.